workflow/artifacts/http/http.go

```diff
@@ -77,39 +81,64 @@ func (h *ArtifactDriver) Save(path string, outputArtifact *wfv1.Artifact) error
 	if err != nil {
 		return err
 	}
-	var req *http.Request
+	defer f.Close()
+	reader := bufio.NewReader(f)
+
 	var url string
 	if outputArtifact.Artifactory != nil && outputArtifact.HTTP == nil {
 		url = outputArtifact.Artifactory.URL
-		req, err = http.NewRequest(http.MethodPut, url, f)
+	} else {
+		url = outputArtifact.HTTP.URL
+	}
+
+	res, err := h.doRequest(http.MethodPut, url, reader, outputArtifact)
+	if err != nil {
+		return err
+	}
+	defer res.Body.Close()
+
+	if res.StatusCode == 307 && outputArtifact.HTTP.FollowTemporaryRedirects {
```

Comment:
I think we should follow redirects always, I don't know when you'd not want to.

Reply:
i was not completely sure about that. Normally, the go client should follow redirects automatically, however, for these 307 it seems it doesnt follow (maybe there are some security concerns?). Thats why introduced the flag, to keep the "normal" behavior.

Reply:
https://github.com/golang/go/issues/7912

Reply:
thanks for pointing me to this. I finally found out that i have to manually define the `GetBody func() (io.ReadCloser, error)` for the redirect to work. I think, we can then also get rid of the `followRedirectFlag`.